packages/components/src/inline-edit/hook.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, negate, noop, flow } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const cancelEvent = ( event ) => (
+	event.preventDefault(), event.stopPropagation(), event
+);
+
+/**
+ * @param {import("react").ChangeEvent<HTMLInputElement>} event
+ */
+const getEventValue = ( { target: { value } } ) => value;
+
+/**
+ * @param {Function[]} handlers
+ */
+const mergeEvent = ( ...handlers ) =>
+	/**
+	 * @param {import("react").ChangeEvent<HTMLInputElement>} event
+	 */
+	( event ) => handlers.forEach( ( handler = noop ) => handler( event ) );
+
+/**
+ * @typedef Props
+ * @property {(value: string) => boolean} validate predicate
+ * @property {( value: string ) => void} onWrongInput called when a validate predicate fails
+ * @property {( value: string ) => void} onCommit called on enter/blur
+ * @property {string} value input value
+ */
+
+/**
+ * @param {Props} props
+ */
+export default function useInlineEdit( {
+	validate = negate( isUndefined ),
+	onWrongInput = noop,
+	onCommit = noop,
+	value: propValue,
+} ) {
+	const [ isInEditMode, setIsInEditMode ] = useState( false );
+	const [ editingValue, setEditingValue ] = useState( propValue );
+	/** @type {import('react').RefObject<HTMLInputElement>} */
+	const inputRef = useRef( null );
+	/** @type {import('react').RefObject<HTMLInputElement>} */
+	const toggleRef = useRef( null );
+	const isInvalid = negate( validate );
+	const changeToEditMode = () => setIsInEditMode( true );
+	const changeToToggleMode = () => setIsInEditMode( false );
+
+	useEffect( () => {
+		setEditingValue( propValue );
+		if ( isInvalid( value ) ) onWrongInput( value );
+	}, [ propValue ] );
+
+	useEffect( () => {
+		if ( isInEditMode ) {
+			inputRef.current?.focus();
+			inputRef.current?.select();
+			setEditingValue( propValue );
+		} else {
+			toggleRef.current?.focus();
+		}
+	}, [ isInEditMode ] );
```

Comment:
Also here `propValue` is not in the deps. Is this intentional?

Reply:
I have simplified this in a recent commit